File: iteration2/CSE3063F21P1_GRP16/src/main/java/Models/E4NoEnoughCredits.java

package Models;

import java.util.ArrayList;

public class E4NoEnoughCredits  implements CourseEvents {
	
	
private ArrayList<Student> students;


public ArrayList<Student> getStudents() {
	return students;
}

public  E4NoEnoughCredits() {
	students=new ArrayList<>();
	
}

@Override
public String toString() {

	String s="    Total Number of Student who couldn't take the course because of they don't have enough credits: "+students.size();
	if(students.size()>0)
		s+="\n     -List Of Students:\n";
	else
		s+="\n";

	for(int i=0;i<students.size();i++) {
		s+="      *"+students.get(i).getFirstName()+" "+students.get(i).getLastName()+"\n";


	}
	return s;
}

}

File: iteration3/CSE3063F21P1_GRP16/models/E4NoEnoughCredits.py
from typing import List

from CourseEvents import CourseEvents

from Student import Student


class E4NoEnoughCredits(CourseEvents):
    def __init__(self):
        self._students = []
    
    def getStudents(self):
        return self._students
    
    def toString(self):
        s = "    Total Number of Student who couldn't take the course because of they don't have enough credits: {}".format(len(self._students))
        if self._students:
            s += "\n     -List Of students:\n"
        else:
            s += "\n"
        for i in range(len(self._students)):
            s += "      *{} {}\n".format(self._students[i].getFirstName(), self._students[i].getLastName())
        return s
